.github/workflows/build-verification.yml

```diff
@@ -20,3 +20,40 @@ jobs:
           develocity-access-key: ${{ secrets.DV_SOLUTIONS_ACCESS_KEY }}
       - name: Build with Gradle
         run: ./gradlew build -x signPluginMavenPublication -i -Porg.gradle.java.installations.auto-download=false
+
+  selftest:
+    name: Publish to Maven Local and Test with Locally Published Plugin
+    runs-on: ubuntu-latest
+    needs: verification
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+      - name: Set up Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          develocity-access-key: ${{ secrets.DV_SOLUTIONS_ACCESS_KEY }}
+      - name: Disable signing on CI
+        run: sed -i 's/isRequired = providers.environmentVariable("CI").isPresent/isRequired = false/' build.gradle.kts
+      - name: Publish to maven local
+        id: publish_to_maven_local
+        run: ./gradlew clean publishToMavenLocal -Dgradle.cache.remote.push=false
+      - name: Modify the project to use the locally published plugin
+        run: |
+          sed -i '1i\
+            pluginManagement {\
+            repositories {\
+            mavenLocal()\
+            gradlePluginPortal()\
+            mavenCentral()\
+          }}\
+          ' settings.gradle.kts
+          sed -i -E 's/(id\("com.gradle.common-custom-user-data-gradle-plugin"\) version )"2[0-9\.]*"/\1"2+"/' settings.gradle.kts
+
+      - name: Run a build with the locally published plugin
```

Comment:
Is the idea to just verify the build succeeds with CCUD applied? You don't need to run a full build to do that.